pkg/segment/writer/stats/segstats.go

```diff
@@ -272,6 +278,29 @@ func AddSegStatsStr(segstats map[string]*SegStats, cname string, strVal string,
 		}
 	}
 
+	CVal := CValueEnclosure{Dtype: SS_DT_STRING, CVal: strVal}
+	if stats.StringStats.Min.Dtype == SS_DT_STRING {
+		res, err := ReduceMinMax(stats.StringStats.Min, CVal, true)
+		if err != nil {
+			log.Errorf("AddSegStatsStr: Error while reducing min: %v", err)
+		} else {
+			stats.StringStats.Min = res
+		}
+	} else {
+		stats.StringStats.Min = CVal
+	}
+
+	if stats.StringStats.Max.Dtype == SS_DT_STRING {
```

Comment:
nit: it would be nice if you can reduce this duplication with the block above for Min, but it's not necessary if you can't find a good way to do that while keeping the code clear

Reply:
I am not able to find a good way to do this and I think we can do it in future, since, the Min/Max of SegStats is anyways going to be changed when we add min/max for strings and we would just combine those as a single CVal instead of Min in both NumStats and StringStats.

Reply:
Sounds good